### PR TITLE
Prefer the URL used to create the client object instead of configuration

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
When the `RawSynchronousFlyteClient` is created, it's given a host.  We should prefer to use this instead of retrieving from configuration again.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Save the url passed into the client instead and when it comes time to get the Admin endpoints, use it first.  Configuration fallback has been left in there in case we make the client url optional down the road.

## Tracking Issue
https://github.com/lyft/flyte/issues/368

## Follow-up issue
NA

### CODE REVIEW

File: flytekit/clis/auth/credentials.py [hunk 1]
```diff
@@ -11,10 +11,8 @@
 )
 from flytekit.configuration.platform import URL as _URL, INSECURE as _INSECURE, HTTP_URL as _HTTP_URL
 
-try:  # Python 3
-    import urllib.parse as _urlparse
-except ImportError:  # Python 2
```
Comment (katrogan) [inline flytekit/clis/auth/credentials.py:16]: bye!

File: flytekit/clis/flyte_cli/main.py [hunk 2]
```diff
@@ -454,7 +454,7 @@ def _render_schedule_expr(lp):
     help="[Optional] Set the flag if you want to list the urns only"
 )
 _project_identifier_option = _click.option(
-    '--identifier',
+    '-p', '--identifier',
```
Comment (EngHabu) [inline flytekit/clis/flyte_cli/main.py:457]: ❤️ 

CC @katrogan is this ok?

File: tests/flytekit/unit/clients/test_raw.py [hunk 3]
```diff
@@ -31,5 +31,7 @@ def test_refresh_credentials_basic(mock_credentials_access, mock_requests):
     os.environ[_CREDENTIALS_SECRET.env_var] = "asdf12345"
 
     mock_client = mock.MagicMock()
+    mock_client.url.return_value = 'flyte.corp.net'
```
Comment (EngHabu) [inline tests/flytekit/unit/clients/test_raw.py:34]: I know that looks cool but I've heard horror stories before about URLs used in tests getting abused by someone buying the domain...

```suggestion
    mock_client.url.return_value = 'flyte.localhost'
```

File: flytekit/__init__.py [hunk 4]
```diff
@@ -2,4 +2,4 @@
 
 import flytekit.plugins
 
-__version__ = '0.10.5'
+__version__ = '0.10.6'
```
Comment (EngHabu) [inline flytekit/__init__.py:5]: I like how you've become ok with this ;)

File: flytekit/clis/flyte_cli/main.py (hunk 2 above)
Comment (katrogan) [reply flytekit/clis/flyte_cli/main.py:457]: of course!

**Comment by bnsblue** [comment]
@wild-endeavor Yes please! When do you expect to merge this fix? I got trapped on this issue for two hours today .
